spec/acceptance/services_controller_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+resource 'Services' do
+  post '/services/register' do
+    example 'Creating and returning a nested data structure' do
+      expect{do_request}
+      .to change{Service.count}
+      .by 1
+      body = JSON.parse response_body
+      expect(body).not_to be_empty
```

Comment:
Good start - some more specific expectations would be good once the controller changes are made.